mautic/mautic
# Guide users to build optimized segments

MAUT-11343 : Guide users to build optimized segments



| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🟢 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description

Add warnings when users are using the below operators 
 **Like :** This Like filter will slow down instance performance. We recommend using “Starts with” if possible.
 **Contains :** This Contains filter will slow down instance performance. We recommend using “Starts with” if possible.
 **Regexp :** This Regexp filter will slow down instance performance. We recommend using “Starts with” if possible.
 **End with :** This Ends With filter will slow down instance performance. We recommend using “Starts with” if possible.

**Field: email**
If any of the above operators is combined with the “Email” field, add another message: Consider using the “Email Domain” filter instead of “Email” to be able to match the email domain. 







---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a segment
3. select filters which are mentioned in description
4. Verify alert message text based on field and filter defined in description
5. verify alert message should be persist on save and edit
